### PR TITLE
serverless: API Gateway through Kong event detection

### PR DESCRIPTION
**Pull request by Hellzy**
### What does this PR do?

Revert https://github.com/DataDog/datadog-agent/pull/19569 and rather introduce a new invocation event detection condition at the end to try to ultimately check if this is a Kong API Gateway event. As of Kong's [plugin](https://github.com/Kong/kong/blob/master/kong/plugins/aws-lambda/request-util.lua#L248-L260), the payload is an API Gateway event one with some missing fields making our `isAPIGatewayEvent` function fail due to the missing `requestContext` and `stage` fields. For this reason we decided to introduce a separate condition to match Kong's.

### Motivation

Rework the failed attempt to support Kong API Gateway events in the previous release.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/trigger/events.go [hunk 1]
```diff
@@ -167,7 +174,7 @@ func Unmarshal(payload []byte) (map[string]any, error) {
 }
 
 func isAPIGatewayEvent(event map[string]any) bool {
-	return json.GetNestedValue(event, "requestcontext") != nil &&
+	return json.GetNestedValue(event, "requestcontext", "stage") != nil &&
```
Comment (purple4reina) [inline pkg/serverless/trigger/events.go:177]: Is this change necessary? I'm struggling to understand why we need to make this check more restrictive.

Comment (Julio-Guerra) [reply pkg/serverless/trigger/events.go:177]: it's a revert of the change we did in the previous release in https://github.com/DataDog/datadog-agent/pull/19569 to restore it the way it was

**Review by purple4reina — APPROVED** [review]

**Review by Julio-Guerra — APPROVED** [review]
